contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -1448,6 +1449,51 @@ TdsTypeUIDToDatum(StringInfo buf)
 	PG_RETURN_POINTER(uuid);
 }
 
+/* Helper Function to convert Spatial Type values into Datum. */
+Datum
+TdsTypeSpatialToDatum(StringInfo buf)
+{
+	bytea	   *result;
+	int			nbytes;
+	int 		loopVar;
+
+	/*
+	 * Here the incoming buf format is -> 4 Byte SRID + 2 Byte Geometry Type + (16 Bytes)*npoints
+	 * But Driver expects -> 4 Byte SRID + 4 Byte Type + 4 Byte npoints + (16 Bytes)*npoints
+	 * so we first shift (16 Bytes)*npoints by 6 Bytes and handle the remaining in between bytes
+	 */
+	for (loopVar = buf->len - 1; loopVar >= buf->cursor + 6; loopVar--) {
+		buf->data[loopVar + 6] = buf->data[loopVar];
+	}
+
+	nbytes = buf->len - buf->cursor + 6;
+	result = (bytea *) palloc0(nbytes + VARHDRSZ);
+	SET_VARSIZE(result, nbytes + VARHDRSZ);
+
+	/* Here we are handling the 8 bytes (4 Byte Type + 4 Byte npoints) which driver expects for 2-D point */
+	if (buf->data[buf->cursor + 4] == 1 && buf->data[buf->cursor + 5] == 12) {
+		
+		buf->data[buf->cursor + 4] = 0x01;
+		buf->data[buf->cursor + 5] = 0x00;
+		buf->data[buf->cursor + 6] = 0x00;
+		buf->data[buf->cursor + 7] = 0x00;
+
+		buf->data[buf->cursor + 8] = 0x01;
+		buf->data[buf->cursor + 9] = 0x00;
+		buf->data[buf->cursor + 10] = 0x00;
+		buf->data[buf->cursor + 11] = 0x00;
```

Comment:
Agree, we shouldn't rely on StringInfo underlying implementation, as it could change in future releases. Maybe use `appendStringInfoBinary` or `appendStringInfoChar`

Reply:
Done.